### PR TITLE
DOC: update the pandas.DateTimeArrayMixin.is_month_start docstring

### PR DESCRIPTION
**Pull request by CalmDownKarm**
- [x] closes #20146 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
should close PR #20146

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -1011,6 +1011,30 @@ def date(self):
         'is_month_start',
         'is_month_start',
         "Logical indicating if first day of month (defined by frequency)")
+    """
+        Indicates whether the date is the first day of the month.
+        Returns
+        -------
+        Series of bool.
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: Can you remove the period at the end, and add a description indented in the next line.

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -1011,6 +1011,30 @@ def date(self):
...
+
+        See Also
+        --------
+        is_month_end : Return a boolean indicating whether the date is
+            the last day of the month.
+        Examples
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: Missing blank line before `Examples`.

**Review by datapythonista — CHANGES_REQUESTED** [review]
There are few formatting problems. There are two things you should do:
- Validate the docstring for formatting errors: `./scripts/validate_docstrings.py pandas.DateTimeArrayMixin.is_month_start`
- Generate the html and see if it renders correctly: `./doc/make.py html --single=pandas.DateTimeArrayMixin.is_month_start`

**Comment by CalmDownKarm** [comment]
@datapythonista I made the changes you asked for, but both commands you asked me to run failed - AttributeError: module 'pandas' has no attribute 'DateTimeArrayMixin' probably a broken setup on my end?

File: pandas/core/arrays/datetimes.py [hunk 3]
```diff
@@ -1011,6 +1011,34 @@ def date(self):
...
         'is_month_start',
         "Logical indicating if first day of month (defined by frequency)")
+    """
+        Indicates whether the date is the first day of the month.
+        
+        Returns
+        -------
+        Series of bool
+            Series of boolean values indicating whether the date is the first
+            day of the month
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: Can you add a period at the end

**Review by datapythonista — COMMENTED** [review]
The command to run is `./scripts/validate_docstrings.py pandas.core.arrays.datetimes.DateTimeArrayMising`.

Got the full path to the class from the PR title, but seems like it's been shortened there.

**Comment by datapythonista** [comment]
is the validation script happy now?

**Comment by datapythonista** [comment]
seems like there are whitespaces at the end of some lines in your code (check travis), can you remove them and validate pep8

**Comment by CalmDownKarm** [comment]
@datapythonista fixed flake8 and validation script works now, but I'm unable to generate the html. 

python scripts/validate_docstrings.py pandas.core.arrays.datetimes.DatetimeArrayMixin.is_month_start
validates fine
but if I run 
```python doc/make.py html --single=pandas.core.arrays.datetimes.DatetimeArrayMixin.is_month_start``` 
I get
AttributeError: module 'pandas.core.arrays' has no attribute 'datetimes'

File: pandas/core/arrays/datetimes.py (hunk 3 above)
Comment (CalmDownKarm) [reply pandas/core/arrays/datetimes.py]: @datapythonista done. Can this be merged/closed?

**Review by datapythonista — APPROVED** [review]
lgtm

I didn't realize before, but probably it makes sense to have a common template for `is_month_start` and `is_month_end`, but we can take care of that in a separate PR.

@jbrockmendel do you want to take a look and merge if you're happy

File: pandas/core/arrays/datetimes.py [hunk 4]
```diff
@@ -1007,10 +1007,39 @@ def date(self):
...
         "The number of days in the month")
     daysinmonth = days_in_month
+    _is_month_start_doc = """
+        Indicates whether the date is the first day of the month.
+
+        Returns
+        -------
+        Series of bool
+            Series of boolean values indicating whether the date is the first
+            day of the month.
```
Comment (jbrockmendel) [inline pandas/core/arrays/datetimes.py]: This generally looks good, only issue is that it isn't always a Series.  On DatetimeIndex it returns a numpy array.

Comment (datapythonista) [reply pandas/core/arrays/datetimes.py]: True. @CalmDownKarm you can take a look at `is_month_end` where this is correct. And as I said, if you want to merge both docstrings into one that would be great. It'd be something like:
```
is_month_doc = """
Indicates whether the date is the {first_or_last} day of the month.
[...]
"""
is_month_start = _field_accessor('is_month_start', 'is_month_start', is_month_doc.format(first_or_last='first'))
```

Comment (CalmDownKarm) [reply pandas/core/arrays/datetimes.py]: fixed and combined. @datapythonista @jbrockmendel

File: pandas/core/arrays/datetimes.py [hunk 5]
```diff
@@ -998,48 +998,62 @@ def date(self):
...
         2   2018-03-01
         dtype: datetime64[ns]
-        >>> dates.dt.is_month_end
-        0    False
-        1    True
-        2    False
+        >>> s.dt.is_month_{start_or_end}
+        0    {FT[0]}
+        1    {FT[1]}
+        2    {FT[2]}
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: I find this a bit too complicated for a docstring. I'd prefer to have both examples (is_month_start and is_month_end) in both docstrings.

File: pandas/core/arrays/datetimes.py [hunk 6]
```diff
@@ -998,48 +998,62 @@ def date(self):
...
+    _is_month_doc = """
+        Indicates whether the date is the {first_or_last} day of the month.
 
         Returns
         -------
         Series or array
-            For Series, returns a Series with boolean values. For
-            DatetimeIndex, returns a boolean array.
+            For Series, returns a Series with bool.
+            For DatetimeIndex, returns a bool array.
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: We use the Python types in the line with the types. In the description the original text was all right.

File: pandas/core/arrays/datetimes.py [hunk 7]
```diff
@@ -998,48 +998,62 @@ def date(self):
...
 
         See Also
         --------
-        is_month_start : Indicator for whether the date is the first day
-            of the month.
+        is_month_{end_or_start} : Return a boolean indicating whether the date
+            is the {last_or_first} day of the month.
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: in the See Also you can also have both methods, so we don't need so many variables.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks!

**Review by datapythonista — APPROVED** [review]
lgtm, thanks!

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @CalmDownKarm !

**Pull request merged**